aif360/algorithms/preprocessing/lfr.py

```diff
@@ -81,33 +85,38 @@ def fit(self, dataset, maxiter=5000, maxfun=5000, **kwargs):
 
         num_train_samples, self.features_dim = np.shape(dataset.features)
 
-        d = np.reshape(
+        protected_attributes = np.reshape(
             dataset.protected_attributes[:, dataset.protected_attribute_names.index(self.protected_attribute_name)],
             [-1, 1])
-        sensitive_idx = np.array(np.where(d == self.unprivileged_group_protected_attribute_value))[0].flatten()
-        nonsensitive_idx = np.array(np.where(d == self.privileged_group_protected_attribute_value))[0].flatten()
-        training_sensitive = dataset.features[sensitive_idx]
-        training_nonsensitive = dataset.features[nonsensitive_idx]
-        ytrain_sensitive = dataset.labels[sensitive_idx]
-        ytrain_nonsensitive = dataset.labels[nonsensitive_idx]
-
-        model_inits = np.random.uniform(size=self.features_dim * 2 + self.k + self.features_dim * self.k)
+        unprivileged_sample_ids = np.array(np.where(protected_attributes == self.unprivileged_group_protected_attribute_value))[0].flatten()
+        privileged_sample_ids = np.array(np.where(protected_attributes == self.privileged_group_protected_attribute_value))[0].flatten()
+        features_unprivileged = dataset.features[unprivileged_sample_ids]
+        features_privileged = dataset.features[privileged_sample_ids]
+        labels_unprivileged = dataset.labels[unprivileged_sample_ids]
+        labels_privileged = dataset.labels[privileged_sample_ids]
+
+        # Initialize the LFR optim objective parameters
+        parameters_initialization = np.random.uniform(size=self.k + self.features_dim * self.k)
         bnd = []
-        for i, _ in enumerate(model_inits):
+        for i, _ in enumerate(parameters_initialization):
             if i < self.features_dim * 2 or i >= self.features_dim * 2 + self.k:
                 bnd.append((None, None))
             else:
                 bnd.append((0, 1))
```

Comment:
I think this should be: `bnd = [(0, 1)]*self.k + [(None, None)]*self.features_dim*self.k` so w is bound to (0, 1) and the prototypes are unbounded.

Reply:
yes. fixed,